cub/cub/agent/agent_find.cuh

```diff
@@ -0,0 +1,250 @@
+// SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#include <cub/config.cuh>
+
+#include <cub/iterator/cache_modified_input_iterator.cuh>
+#include <cub/thread/thread_load.cuh>
+#include <cub/util_arch.cuh>
+#include <cub/util_type.cuh>
+
+#include <cuda/std/__type_traits/integral_constant.h>
+
+CUB_NAMESPACE_BEGIN
+template <int NominalBlockThreads4B,
+          int NominalItemsPerThread4B,
+          typename ComputeT,
+          int VectorLoadLength,
+          CacheLoadModifier _LOAD_MODIFIER,
+          typename ScalingType = cub::detail::MemBoundScaling<NominalBlockThreads4B, NominalItemsPerThread4B, ComputeT>>
+struct AgentFindPolicy : ScalingType
```

Comment:
Critical: agents should not be part of the public API. Please move it to a detail namespace and use `snake_case`.

Suggestion: avoid the agent all together and just move the implementation into `DeviceFindKernel`.